File: haskell/bob/Bob.hs
module Bob (responseFor) where


import Data.Char (isSpace, isLetter, isUpper)


responseFor :: String -> String
responseFor msg
  | all isSpace msg = "Fine. Be that way!"
  | isYelling msg   = "Woah, chill out!"
  | last msg == '?' = "Sure."
  | otherwise       = "Whatever."


isYelling :: String -> Bool
isYelling msg = not (null letters) && all isUpper letters
  where letters = filter isLetter msg

File: javascript/bob/bob.js
var Bob = module.exports = function() {
  
  this.hey = function(message) {

    if (!message.trim()) {
      return 'Fine. Be that way!';

    } else if (message === message.toUpperCase() &&
               message !== message.toLowerCase()) {
      return 'Woah, chill out!';

    } else if (message.slice(-1) === '?') {
      return 'Sure.';

    } else {
      return 'Whatever.';
    };
  };
};
